library(Seurat)
library(tidyverse)
library(patchwork)
library(nichenetr)
library(EnhancedVolcano)
library(GEOquery)
library(hacksig)
library(pROC)
library(UCell)
library(sjmisc)

super.integrated <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Irg1 and iNOS scRNAseq 220315/b6 sp140 irg1 inos v2")
#Note: the celltype annotation in the super.integrated object for Spp1+ IMs is "ISG IM" while the annotation for Trem2+ IM is "Activated IM"

#Graphing the data for Figures.

#Fig. 2a
Idents(super.integrated) <- "celltype"
DimPlot(super.integrated, split.by = "cell_type")

#Fig. 2b - tb.counts.norm, TB.infected and strict genotype signatures generated in Mouse to Human Signatures Clean.R
tb.counts.norm <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/tb.counts.norm")
B6.strict <- read.csv(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/B6 Strict Genes.csv") %>% filter(change == "up")
Sp140.strict <- read.csv(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/Sp140 Strict Genes.csv") %>% filter(change == "up")
Irg1.strict <- read.csv(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/Irg1 Strict Genes.csv") %>% filter(change == "up")
iNOS.strict <- read.csv(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/iNOS Strict Genes.csv") %>% filter(change == "up")

TB.infected <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/TB.infected")

mouse.tb <- hack_sig(tb.counts.norm, signatures = c(list(unique(B6.strict$gene)),list(unique(Sp140.strict$gene)),list(unique(Irg1.strict$gene)),list(unique(iNOS.strict$gene))) , method = "singscore")
mouse.tb$status <- ifelse(grepl(paste(TB.infected, collapse = "|"), mouse.tb$sample_id), "TB","Uninfected")
b6.roc <- roc(mouse.tb$status, mouse.tb$sig1, levels = c("TB","Uninfected"))
sp140.roc <- roc(mouse.tb$status, mouse.tb$sig2, levels = c("TB","Uninfected"))
irg1.roc <- roc(mouse.tb$status, mouse.tb$sig3, levels = c("TB","Uninfected"))
inos.roc <- roc(mouse.tb$status, mouse.tb$sig4, levels = c("TB","Uninfected"))
plot(b6.roc, print.auc = TRUE, col = "red", print.auc.y =0.47)
plot(sp140.roc, add=TRUE, print.auc = TRUE, col = "blue",  print.auc.y =0.4)
plot(irg1.roc,add = TRUE, print.auc = TRUE, col = "chartreuse", print.auc.y =0.33)
plot(inos.roc, add=TRUE, print.auc = TRUE, col = "cadetblue2",  print.auc.y =0.27)

#Fig. 2c The mouse ifnb and ifng signatures are from PMID: 38029747 (generated by Mouse Interferon Signature.R)
cyto.sig <- list()
cyto.sig$mouse_ifnb <- specific.ifnb.mouse$gene
cyto.sig$mouse_ifng <- specific.ifng.mouse$gene

super.integrated <- AddModuleScore_UCell(super.integrated, features = cyto.sig)

super.integrated$mouse_ifn_sig <- ifelse(super.integrated$mouse_ifnb_UCell >= 0.24 & super.integrated$mouse_ifng_UCell >= 0.32, "both", 
                                         ifelse(super.integrated$mouse_ifnb_UCell >= 0.24 & super.integrated$mouse_ifng_UCell < 0.32, "ifnb", 
                                                ifelse(super.integrated$mouse_ifnb_UCell < 0.24 & super.integrated$mouse_ifng_UCell >= 0.32, "ifng","neither")))


Idents(super.integrated) <- "state"
super.mtb <- subset(super.integrated, idents = "Mtb+")
super.naive <- subset(super.integrated, idents = "naive")

Idents(super.mtb) <- "mouse_ifn_sig"
levels(super.mtb) <- c("neither","both","ifng","ifnb")
DimPlot(super.mtb, split.by = "cell_type",cols = c("grey","chartreuse3","red","blue"))

Idents(super.naive) <- "mouse_ifn_sig"
levels(super.naive) <- c("neither","both","ifng","ifnb")
DimPlot(super.naive, cols = c("grey","chartreuse3","red","blue"))

#Fig. 2d - resulting analysis was plotted in Prism to generate the final figure
Idents(super.integrated) <- "state"
super.mtb <- subset(super.integrated, idents = "Mtb+")
Idents(super.mtb) <- "celltype"
simple.mtb <- RenameIdents(super.mtb, 'ISG Mature Neut' = "Neutrophil", 'ISG Activated Neut' = "Neutrophil", 'Aged Neut' = "Neutrophil", 'Mature Neut' = "Neutrophil",
                           'ISG Old Neut' = "Neutrophil", 'ISG Young Neut' = "Neutrophil", 'CD63 Neut' = "Neutrophil", 'Mmp8 Neut' = "Neutrophil",'Nos2 Neut' = "Neutrophil",
                           'Camp Mmp8 Neut' = "Neutrophil", 'CD16-2 Mono' = "Mono", 'ISG IM' = "IM", 'Activated IM' = "IM", 'Activated AM' = "AM")
simple.mtb <- subset(simple.mtb, idents = c("Neutrophil","Mono","IM","AM","DC"))
simple.mtb$old.celltype <- simple.mtb$celltype
simple.mtb$celltype <- Idents(simple.mtb)

simp.genotype <- list()
Idents(simple.mtb) <- "mouse_id"

populations <- as.character(unique(simple.mtb$celltype))
cell_type <- rep(unique(simple.mtb$cell_type),3)

for(z in 1:3){
  a <- subset(simple.mtb, idents = z)
  Idents(a) <- "cell_type"
  for(i in 1:4){
    y <- assign(paste("simp.", cell_type[i], sep = ""), subset(a, idents = cell_type[i]))
    simp.genotype[[length(simp.genotype)+1]] <- y
    Idents(simp.genotype[[length(simp.genotype)]]) <- "celltype"
  }
}


#Missing populations
#1 - AM, DC
#3 - DC
#7 - DC
#11 - DC
cell_type.pop <- list()
for(i in 1:3){
  for(j in 1:12){
    y <- assign(paste("simp", cell_type[j], populations[i], sep = "."), subset(simp.genotype[[j]], idents = populations[i]))
    cell_type.pop[[length(cell_type.pop)+1]] <- y
  }
}

cyt.sig.total <- lapply(cell_type.pop, function(x) {(prop.table(table(Idents(x), x$mouse_ifn_sig)))})
cyt.sig.total <- lapply(cyt.sig.total, function(x) {as.data.frame.matrix(x)})
cyt.sig.total <- data.table:::rbindlist(cyt.sig.total, fill=TRUE)
cyt.sig.total[is.na(cyt.sig.total)] = 0
cyt.sig.total <- cyt.sig.total[]*100
cyt.sig.total$mouse.id <- rep(c(rep(1,4),rep(2,4),rep(3,4)),3)
cyt.sig.total$genotype <- c(rep(cell_type,3))
cyt.sig.total$celltype <- c(rep("IM",12),rep("Neutrophil",12),rep("Monocyte",12))
write.csv(cyt.sig.total, file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/Plots 230806/IFN Response IM Neut Mono.csv")

#For AM exclude 1
cell_type.pop <- list()
for(i in 4){
  for(j in c(2:12)){
    y <- assign(paste("simp", cell_type[j], populations[i], sep = "."), subset(simp.genotype[[j]], idents = populations[i]))
    cell_type.pop[[length(cell_type.pop)+1]] <- y
  }
}

cyt.sig.total <- lapply(cell_type.pop, function(x) {(prop.table(table(Idents(x), x$mouse_ifn_sig)))})
cyt.sig.total <- lapply(cyt.sig.total, function(x) {as.data.frame.matrix(x)})
cyt.sig.total <- data.table:::rbindlist(cyt.sig.total, fill=TRUE)
cyt.sig.total[is.na(cyt.sig.total)] = 0
cyt.sig.total <- cyt.sig.total[]*100
cyt.sig.total$mouse.id <- c(rep(1,3),rep(2,4),rep(3,4))
cyt.sig.total$genotype <- c("Sp140","Irg1","iNOS",rep(c("B6","Sp140","Irg1","iNOS"),2))
cyt.sig.total$celltype <- c(rep("AM",11))
write.csv(cyt.sig.total, file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/Plots 230806/IFN Response AM.csv")

#Supp. Fig. 1a
Idents(super.integrated) <- "celltype"
levels(super.integrated) <- c("gCap","aCap","fibroblast","Stromal?","DC","B cell","Basophil","Camp Mmp8 Neut","Mmp8 Neut","Mature Neut",
                              "Aged Neut","ISG Mature Neut","ISG Activated Neut","ISG Young Neut","ISG Old Neut","Nos2 Neut","CD63 Neut",
                              "AM","Activated AM","Cycling AM","CD16-2 Mono","Mono","IM","Activated IM","ISG IM")
DotPlot(super.integrated, features = c("Ptprc","Cd19","Fcer1a","adt_Ly6G","adt_MHCII","Itgax","adt_SiglecF","Fcgr1","Mertk","Ccr2","Mki67"))

#Supp. Fig. 1b
super.neut <- subset(super.integrated, idents = c("Camp Mmp8 Neut","Mmp8 Neut","Mature Neut","Aged Neut","ISG Mature Neut",
                                                  "ISG Activated Neut","ISG Young Neut","ISG Old Neut","Nos2 Neut","CD63 Neut"))
DotPlot(super.neut, features = c("Camp","Mmp8","Cxcr2","Cxcr4","Isg15","Nos2","Cd63"))

#Supp. Fig. 1c
Idents(super.integrated) <- "state"
super.naive <- subset(super.integrated, idents = "naive")
DimPlot(super.naive, split.by = "cell_type")

#Supp. Fig. 1d - differentially expressed genes calculated here with the final visual generated in Adobe Illustrator
Idents(super.integrated) <- "state"
naive.integrated <- subset(super.integrated,idents = "naive")
Idents(naive.integrated) <- "celltype"
naive.integrated <- RenameIdents(naive.integrated, 'ISG Mature Neut' = "Neutrophil", 'ISG Activated Neut' = "Neutrophil", 'Aged Neut' = "Neutrophil", 'Mature Neut' = "Neutrophil",
                                 'ISG Old Neut' = "Neutrophil", 'ISG Young Neut' = "Neutrophil", 'CD63 Neut' = "Neutrophil", 'Mmp8 Neut' = "Neutrophil",'Nos2 Neut' = "Neutrophil",
                                 'Camp Mmp8 Neut' = "Neutrophil", 'CD16-2 Mono' = "Mono", 'ISG IM' = "IM", 'Activated IM' = "IM", 'Activated AM' = "AM")

#Make Venn Diagrams for B6 vs Sp140; B6 vs Nos2; B6 vs Acod1 for IM, AM, Mono, and Neutrophils for naive mice
myCol <- brewer.pal(3, "Pastel2")

#Neutrophils
naive.neutrophil <- subset(naive.integrated, idents = "Neutrophil")
Idents(naive.neutrophil) <- "cell_type"
naive.neutrophils.sp140 <- FindMarkers(naive.neutrophil, ident.1 = "B6",ident.2 = "Sp140")
naive.neutrophils.nos2 <- FindMarkers(naive.neutrophil, ident.1 = "B6",ident.2 = "iNOS")
naive.neutrophils.acod1 <- FindMarkers(naive.neutrophil, ident.1 = "B6",ident.2 = "Irg1")
naive.neutrophils.sp140 <- naive.neutrophils.sp140 %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
naive.neutrophils.nos2 <- naive.neutrophils.nos2 %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
naive.neutrophils.acod1 <- naive.neutrophils.acod1 %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
venn.diagram(x = list(rownames(naive.neutrophils.sp140),rownames(naive.neutrophils.nos2),rownames(naive.neutrophils.acod1)),
             category.names = c("Sp140", "Nos2","Acod1"), filename = "naive neutrophils.tiff", resolution = 600, 
             col=myCol,
             fill = c(alpha(myCol[1],0.3), alpha(myCol[2],0.3), alpha(myCol[3],0.3)),
             euler.d = FALSE,
             scaled = FALSE,
             cat.cex = 0)

#AM
naive.AM <- subset(naive.integrated, idents = "AM")
Idents(naive.AM) <- "cell_type"
naive.AM.sp140 <- FindMarkers(naive.AM, ident.1 = "B6",ident.2 = "Sp140")
naive.AM.nos2 <- FindMarkers(naive.AM, ident.1 = "B6",ident.2 = "iNOS")
naive.AM.acod1 <- FindMarkers(naive.AM, ident.1 = "B6",ident.2 = "Irg1")
naive.AM.sp140 <- naive.AM.sp140  %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
naive.AM.nos2 <- naive.AM.nos2 %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
naive.AM.acod1 <- naive.AM.acod1 %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
venn.diagram(x = list(rownames(naive.AM.sp140),rownames(naive.AM.nos2),rownames(naive.AM.acod1)),
             category.names = c("Sp140", "Nos2","Acod1"), filename = "naive AM.tiff", resolution = 600, 
             col=myCol,
             fill = c(alpha(myCol[1],0.3), alpha(myCol[2],0.3), alpha(myCol[3],0.3)),
             euler.d = FALSE,
             scaled = FALSE,
             cat.cex = 0)

#IM
naive.IM <- subset(naive.integrated, idents = "IM")
Idents(naive.IM) <- "cell_type"
naive.IM.sp140 <- FindMarkers(naive.IM, ident.1 = "B6",ident.2 = "Sp140")
naive.IM.nos2 <- FindMarkers(naive.IM, ident.1 = "B6",ident.2 = "iNOS")
naive.IM.acod1 <- FindMarkers(naive.IM, ident.1 = "B6",ident.2 = "Irg1")
naive.IM.sp140 <- naive.IM.sp140  %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
naive.IM.nos2 <- naive.IM.nos2  %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
naive.IM.acod1 <- naive.IM.acod1  %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
venn.diagram(x = list(rownames(naive.IM.sp140),rownames(naive.IM.nos2),rownames(naive.IM.acod1)),
             category.names = c("Sp140", "Nos2","Acod1"), filename = "naive IM.tiff", resolution = 600, 
             col=myCol,
             fill = c(alpha(myCol[1],0.3), alpha(myCol[2],0.3), alpha(myCol[3],0.3)),
             euler.d = FALSE,
             scaled = FALSE,
             cat.cex = 0)

#Monocytes
naive.mono <- subset(naive.integrated, idents = "Mono")
Idents(naive.mono) <- "cell_type"
naive.mono.sp140 <- FindMarkers(naive.mono, ident.1 = "B6",ident.2 = "Sp140")
naive.mono.nos2 <- FindMarkers(naive.mono, ident.1 = "B6",ident.2 = "iNOS")
naive.mono.acod1 <- FindMarkers(naive.mono, ident.1 = "B6",ident.2 = "Irg1")
naive.mono.sp140 <- naive.mono.sp140 %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
naive.mono.nos2 <- naive.mono.nos2  %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
naive.mono.acod1 <- naive.mono.acod1  %>% filter(avg_log2FC > 1 | avg_log2FC < -1) %>% filter(p_val_adj < 0.05)
venn.diagram(x = list(rownames(naive.mono.sp140),rownames(naive.mono.nos2),rownames(naive.mono.acod1)),
             category.names = c("Sp140", "Nos2","Acod1"), filename = "naive monocytes.tiff", resolution = 600, 
             col=myCol,
             fill = c(alpha(myCol[1],0.3), alpha(myCol[2],0.3), alpha(myCol[3],0.3)),
             euler.d = FALSE,
             scaled = FALSE,
             cat.cex = 0)

#Sup. Fig. 2a - strict genotype signatures generated in Mouse to Human Signatures Clean.R; heatmap further annotated for immunosuppressive genes in Adobe Illustrator
B6.strict <- read.csv(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/B6 Strict Genes.csv") %>% filter(change == "up")
Sp140.strict <- read.csv(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/Sp140 Strict Genes.csv") %>% filter(change == "up")
Irg1.strict <- read.csv(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/Irg1 Strict Genes.csv") %>% filter(change == "up")
iNOS.strict <- read.csv(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/iNOS Strict Genes.csv") %>% filter(change == "up")

B6.strict <- B6.strict %>% filter(avg_log2FC > 3)  %>%  arrange(avg_log2FC) %>% dplyr::select(avg_log2FC, gene)
Sp140.strict <- Sp140.strict %>% filter(avg_log2FC > 3)  %>%  arrange(avg_log2FC) %>% dplyr::select(avg_log2FC, gene)
Irg1.strict <- Irg1.strict %>% filter(avg_log2FC > 3)  %>%  arrange(avg_log2FC) %>% dplyr::select(avg_log2FC, gene)
iNOS.strict <- iNOS.strict %>% filter(avg_log2FC > 3)  %>%  arrange(avg_log2FC) %>% dplyr::select(avg_log2FC, gene)

length(unique(c(Sp140.strict$gene,Irg1.strict$gene,iNOS.strict$gene,B6.strict$gene))) #43
all.heatmap.genes <- data.frame(avg_log2FC = rep(-1, 43), gene = unique(c(Sp140.strict$gene,Irg1.strict$gene,iNOS.strict$gene,B6.strict$gene)))

B6.strict <- rbind(B6.strict, filter(all.heatmap.genes, gene %in% setdiff(all.heatmap.genes$gene, B6.strict$gene))) %>%  arrange(avg_log2FC)
B6.strict$genotype <- "B6"
Sp140.strict <- rbind(Sp140.strict, filter(all.heatmap.genes, gene %in% setdiff(all.heatmap.genes$gene, Sp140.strict$gene))) %>%  arrange(avg_log2FC)
Sp140.strict$genotype <- "Sp140"
Irg1.strict <- rbind(Irg1.strict, filter(all.heatmap.genes, gene %in% setdiff(all.heatmap.genes$gene, Irg1.strict$gene))) %>%  arrange(avg_log2FC)
Irg1.strict$genotype <- "Irg1"
iNOS.strict <- rbind(iNOS.strict, filter(all.heatmap.genes, gene %in% setdiff(all.heatmap.genes$gene, iNOS.strict$gene))) %>%  arrange(avg_log2FC)
iNOS.strict$genotype <- "iNOS"
gene.sig.combined <- rbind(B6.strict,Sp140.strict,Irg1.strict,iNOS.strict) %>% arrange(gene)

ggplot(gene.sig.combined, aes(genotype, gene, fill= avg_log2FC)) + 
  geom_tile() +
  scale_fill_gradient2(low = "black")

#Sup. Fig. 2b - berry.southafrica.tb and berry.southafrica.norm are from Human TB Signatures_clean.R
berry.southafrica.tb <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/berry.southafrica.tb")
berry.southafrica.norm<- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/berry.southafrica.norm")

berry.southafrica.sig <- hack_sig(berry.southafrica.norm, signatures = c(list(B6.strict$gene),list(Sp140.strict$gene),list(Irg1.strict$gene),list(iNOS.strict$gene)), method = "singscore")
berry.southafrica.sig$status <- ifelse(grepl(paste(berry.southafrica.tb, collapse = "|"), berry.southafrica.sig$sample_id), "TB", "LTBI")
b6.berry.southafrica.roc <- roc(berry.southafrica.sig$status, berry.southafrica.sig$sig1, levels = c("TB","LTBI"))
sp140.berry.southafrica.tb.roc <- roc(berry.southafrica.sig$status, berry.southafrica.sig$sig2, levels = c("TB","LTBI"))
irg1.berry.southafrica.tb.roc <- roc(berry.southafrica.sig$status, berry.southafrica.sig$sig3, levels = c("TB","LTBI"))
inos.berry.southafrica.tb.roc <- roc(berry.southafrica.sig$status, berry.southafrica.sig$sig4, levels = c("TB","LTBI"))
plot(b6.berry.southafrica.roc, print.auc = TRUE, col = "red", print.auc.y =0.47)
plot(sp140.berry.southafrica.tb.roc, add=TRUE, print.auc = TRUE, col = "blue",  print.auc.y =0.4)
plot(irg1.berry.southafrica.tb.roc,add = TRUE, print.auc = TRUE, col = "chartreuse", print.auc.y =0.33)
plot(inos.berry.southafrica.tb.roc, add=TRUE, print.auc = TRUE, col = "cadetblue2",  print.auc.y =0.27)

#Fig. 3a
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
#remove weirdo outliers
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "celltype"
FeaturePlot(super.infected.IM_filtered, features = c("Spp1","Trem2","Cxcl9"))

#Fig. 3b
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
#remove weirdo outliers
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "celltype"
DimPlot(super.infected.IM_filtered)

#Fig. 3c
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
#remove weirdo outliers
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "celltype"
DimPlot(super.infected.IM_filtered, split.by = "cell_type")

#Fig. 3d
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "celltype"
DimPlot(super.infected.IM_filtered, split.by = "cell_type")
DimPlot(super.infected.IM_filtered, split.by = "state")
Idents(super.infected.IM_filtered) <- "state"
super.infected.IM_filtered.mtbpos <- subset(super.infected.IM_filtered, idents = "Mtb+")
Idents(super.infected.IM_filtered.mtbpos) <- "celltype"
simple.IM.mtb <- subset(super.infected.IM_filtered.mtbpos, idents = c("Activated IM","ISG IM","IM"))
simple.IM.mtb$old.celltype <- simple.IM.mtb$celltype
simple.IM.mtb$celltype <- Idents(simple.IM.mtb)

simp.genotype <- list()
Idents(simple.IM.mtb) <- "mouse_id"

populations <- as.character(unique(simple.IM.mtb$celltype))
cell_type <- rep(unique(simple.IM.mtb$cell_type),3)

for(z in 1:3){
  a <- subset(simple.IM.mtb, idents = z)
  Idents(a) <- "cell_type"
  for(i in 1:4){
    y <- assign(paste("simp.", cell_type[i], sep = ""), subset(a, idents = cell_type[i]))
    simp.genotype[[length(simp.genotype)+1]] <- y
    Idents(simp.genotype[[length(simp.genotype)]]) <- "celltype"
  }
}

im.total <- lapply(simp.genotype, function(x) {(prop.table(table(Idents(x), x$celltype)))})
im.total <- lapply(im.total, function(x) {as.data.frame.matrix(x)})
im.total <- data.table:::rbindlist(im.total, fill=TRUE)
im.total[is.na(im.total)] = 0
im.total <- im.total[]*100
im.total$mouse.id <- rep(c(rep(1,12),rep(2,12),rep(3,12)))
im.total$genotype <- rep(c(rep("B6",3),rep("Sp140",3),rep("Irg1",3),rep("iNOS",3)),3)
write.csv(im.total, file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Irg1 and iNOS scRNAseq 220315/Plots 240116/Mtb-infected IM Subsets by Genotype.csv")

#Fig. 3e
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "state"
super.infected.IM_filtered.mtbpos <- subset(super.infected.IM_filtered, idents = "Mtb+")
Idents(super.infected.IM_filtered.mtbpos) <- "celltype"
Spp1.IM.genes <- FindMarkers(super.infected.IM_filtered.mtbpos, ident.1 = "ISG IM", ident.2 = "Activated IM")
Spp1.IM.volcano <- Spp1.IM.genes[c(2,5)]
EnhancedVolcano(Spp1.IM.volcano,
                lab = rownames(Spp1.IM.volcano),
                x = 'avg_log2FC',
                y = 'p_val_adj',
                title = "Mtb-infected Spp1 IM vs Trem2 IM",
                subtitle = "",
                drawConnectors = T,
                arrowheads = F,
                selectLab = c("Cxcl9","Spp1","Il1rn","C1qa","C1qc","C1qb","Trem2","Ccl5","Mmp2","Tnfsf13b","Lyz2","Apoe","Ubd","Dnase1l3","Saa3","C3","Cd9","Inhba","Lgals3","Cd53","Upp1","Il7r","Cxcl3","Ccl4","Ccl2","Ccr1","Phlda1","Slfn4","Clec4a1","C1ra","Clec4n","Edn1","Cd24a","Basp1"),
                raster = T)

#Fig. 3f
somewhat.simple <- super.integrated
Idents(somewhat.simple) <- "celltype"
somewhat.simple <- RenameIdents(somewhat.simple, 'ISG Mature Neut' = "Neutrophil", 'ISG Activated Neut' = "Neutrophil", 'Aged Neut' = "Neutrophil", 'Mature Neut' = "Neutrophil",
                                'ISG Old Neut' = "Neutrophil", 'ISG Young Neut' = "Neutrophil", 'CD63 Neut' = "Neutrophil", 'Mmp8 Neut' = "Neutrophil",'Nos2 Neut' = "Neutrophil",
                                'Camp Mmp8 Neut' = "Neutrophil", 'Activated AM' = "AM")
super.somewhat.simple <- subset(somewhat.simple, idents = c("Neutrophil","Mono","IM", "ISG IM", "Activated IM","AM","DC"))
super.somewhat.simple$celltype <- Idents(super.somewhat.simple)
Idents(super.somewhat.simple) <- "infected"
mtb.somewhat <- subset(super.somewhat.simple, idents = "infected")
Idents(mtb.somewhat) <- "celltype"
DimPlot(mtb.somewhat)
mtb.somewhat$celltype <- factor(mtb.somewhat$celltype, levels = c("AM","Mono","IM","Activated IM","ISG IM","Neutrophil"))
VlnPlot(mtb.somewhat, features = "Il1rn", split.by = "cell_type", idents = c("AM","Mono","IM","Activated IM","ISG IM","Neutrophil"), group.by = "celltype")

#Fig. 3g - cd8.nhp from Flynn CD8 NHP_Clean
cd8.nhp <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Flynn Shalek CD8 Depletion NHP data/nhp.rds")
Idents(cd8.nhp) <- "treatment"
cd8.nhp.igg <- subset(cd8.nhp, idents = "IgG")
Idents(cd8.nhp.igg) <- "seurat_clusters"
cd8.nhp.igg <- RenameIdents(cd8.nhp.igg, `0` = "TREM2 IM",`1` = "T cell", `2` = "AM",`3` = "Proliferating T cells", `4` = "SPP1 IM", `5` = "Monocytes",`6` = "Mast",`7` = "T cell",`8` = "NK",`9` = "T cell", `10` = "IM", `11` = "TREM2 IM", `12` = "TREM2 IM", `13` = "B Cells", `14` = "cDC2",`15` = "NKT", 
                            `16` = "T cell",`17` = "Plasma Cells", `18` = "T2P",`19` = "IM", `20` = "cDC1", `21` = "Endothelial", `22` = "Proliferating Macs", `23` = "Fibroblasts",`24` = "T cell", `25` = "pDC", `26` = "Proliferating Mast",`27` = "Neutrophils", `28` = "T1P")
DimPlot(cd8.nhp.igg, label = T)

#Fig. 3h - cd8.nhp from Flynn CD8 NHP_Clean
cd8.nhp <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Flynn Shalek CD8 Depletion NHP data/nhp.rds")
Idents(cd8.nhp) <- "treatment"
cd8.nhp.igg <- subset(cd8.nhp, idents = "IgG")
Idents(cd8.nhp.igg) <- "seurat_clusters"
cd8.nhp.igg <- RenameIdents(cd8.nhp.igg, `0` = "TREM2 IM",`1` = "T cell", `2` = "AM",`3` = "Proliferating T cells", `4` = "SPP1 IM", `5` = "Monocytes",`6` = "Mast",`7` = "T cell",`8` = "NK",`9` = "T cell", `10` = "IM", `11` = "TREM2 IM", `12` = "TREM2 IM", `13` = "B Cells", `14` = "cDC2",`15` = "NKT", 
                            `16` = "T cell",`17` = "Plasma Cells", `18` = "T2P",`19` = "IM", `20` = "cDC1", `21` = "Endothelial", `22` = "Proliferating Macs", `23` = "Fibroblasts",`24` = "T cell", `25` = "pDC", `26` = "Proliferating Mast",`27` = "Neutrophils", `28` = "T1P")
nhp.SPP1.TREM2.genes <- FindMarkers(cd8.nhp.igg, ident.1 = "SPP1 IM", ident.2 = "TREM2 IM")
nhp.SPP1.TREM2.volcano <- nhp.SPP1.TREM2.genes[c(2,5)]
EnhancedVolcano(nhp.SPP1.TREM2.volcano,
                lab = rownames(nhp.SPP1.TREM2.volcano),
                x = 'avg_log2FC',
                y = 'p_val_adj',
                title = "SPP1 IM vs TREM2 IM",
                subtitle = "",
                drawConnectors = T,
                arrowheads = F,
                selectLab = c("SPP1","TREM2","C1QC","C1QB","AXL","IDO1","IL18BP","CD274","IL6","IL1RN","EGLN3","NDRG1","CLEC6A","ADAM8","CHST11", "MMP2","APOC2","CD101","LY86","GPR155","ITGB5","ITGA9","GLO1","CCL18"),
                raster = T)

#Fig. 3i - human.tb from Human TB Lung.R
human.tb <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Human TB resected lung/human.tb")
DimPlot(human.tb, label = T)

#Fig. 3j - human.tb from Human TB Lung.R
human.tb <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Human TB resected lung/human.tb")
human.SPP1.TREM2.genes <- FindMarkers(human.tb, ident.1 = "SPP1 IM", ident.2 = "TREM2 IM")
human.SPP1.TREM2.volcano <- human.SPP1.TREM2.genes[c(2,5)]
EnhancedVolcano(human.SPP1.TREM2.volcano,
                lab = rownames(human.SPP1.TREM2.volcano),
                x = 'avg_log2FC',
                y = 'p_val_adj',
                title = "SPP1 IM vs TREM2 IM",
                subtitle = "",
                drawConnectors = T,
                arrowheads = F,
                selectLab = c("SPP1", "TREM2","IL1RN","IDO1","C1QB","C1QC","C1QA","APOC1","APOE","CCL18","CD274","AXL","IL10","OSM","IL1R2","FCN1","APOBEC3A","MPEG1","SRGN","METRNL","SCIMP","LILRA5","CD93",
                              "CCR2","IGF1","MARCO","CXCL16","LGALS3","IFITM3","CD59","ALDH2","BASP1"),
                raster = T)

#Fig. 3k - human.tb from Human TB Lung.R
human.tb <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Human TB resected lung/human.tb")
myeloid <- subset(human.tb, idents = c("IM", "SPP1 IM", "TREM2 IM", "FOLR2 IM","monocytes"))
levels(myeloid) <- c("SPP1 IM","FOLR2 IM","TREM2 IM","IM","monocytes")
VlnPlot(myeloid, features = "IL1RN", idents = c("monocytes","IM","TREM2 IM", "FOLR2 IM","SPP1 IM"), split.by = "FDG.signal")

#Fig. 3l - tb.counts.norm and TB.infected are from Mouse to Human Signatures Clean.R
tb.counts.norm <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/tb.counts.norm")
TB.infected <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/TB.infected")

#Flip dataframe with normalized counts
counts <- rotate_df(tb.counts.norm)
counts <- counts[1:249,]
counts$sample <- row.names(counts) #23687
counts$Class <- ifelse(grepl(paste(TB.infected, collapse = "|"), counts$sample), "A","B")

#ROC for just IL1RN
IL1RN.score <- counts %>% dplyr::select(c("sample","Class","IL1RN"))
IL1RN.score$status <- ifelse(grepl(paste(TB.infected, collapse = "|"), IL1RN.score$sample), "TB","Uninfected")
IL1RN.score$IL1RN <- as.numeric(IL1RN.score$IL1RN)
IL1RN.roc <- roc(IL1RN.score$status, IL1RN.score$IL1RN)
plot(IL1RN.roc, print.auc = TRUE, col = "red", print.auc.y =0.47)

#Fig. 3m - berry.southafrica.tb and berry.southafrica.norm are from Human TB Signatures_clean.R
berry.southafrica.tb <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/berry.southafrica.tb")
berry.southafrica.norm<- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/berry.southafrica.norm")

berry.southafrica.sig <- hack_sig(berry.southafrica.norm, signatures = c(list(B6.strict$gene),list(Sp140.strict$gene),list(Irg1.strict$gene),list(iNOS.strict$gene)), method = "singscore")
berry.southafrica.sig$status <- ifelse(grepl(paste(berry.southafrica.tb, collapse = "|"), berry.southafrica.sig$sample_id), "TB", "LTBI")

counts <- rotate_df(berry.southafrica.norm)
counts <- counts[1:47,]
counts$sample <- row.names(counts) #23687
counts$Class <- ifelse(grepl(paste(berry.southafrica.tb, collapse = "|"), berry.southafrica.sig$sample_id), "TB", "LTBI")

IL1RN.score.berrySA <- counts %>% dplyr::select(c("sample","Class","IL1RN"))
IL1RN.score.berrySA$status <- ifelse(grepl(paste(berry.southafrica.tb, collapse = "|"), berry.southafrica.sig$sample_id), "TB", "LTBI")
IL1RN.score.berrySA$IL1RN <- as.numeric(IL1RN.score.berrySA$IL1RN)
berrySA.IL1RN.roc <- roc(IL1RN.score.berrySA$status, IL1RN.score.berrySA$IL1RN)
plot(berrySA.IL1RN.roc, print.auc = TRUE, col = "red", print.auc.y =0.47)

#Supp. Fig. 3a - cd8.nhp from Flynn CD8 NHP_Clean
cd8.nhp <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Flynn Shalek CD8 Depletion NHP data/nhp.rds")
Idents(cd8.nhp) <- "treatment"
cd8.nhp.igg <- subset(cd8.nhp, idents = "IgG")
Idents(cd8.nhp.igg) <- "seurat_clusters"
cd8.nhp.igg <- RenameIdents(cd8.nhp.igg, `0` = "TREM2 IM",`1` = "T cell", `2` = "AM",`3` = "Proliferating T cells", `4` = "SPP1 IM", `5` = "Monocytes",`6` = "Mast",`7` = "T cell",`8` = "NK",`9` = "T cell", `10` = "IM", `11` = "TREM2 IM", `12` = "TREM2 IM", `13` = "B Cells", `14` = "cDC2",`15` = "NKT", 
                            `16` = "T cell",`17` = "Plasma Cells", `18` = "T2P",`19` = "IM", `20` = "cDC1", `21` = "Endothelial", `22` = "Proliferating Macs", `23` = "Fibroblasts",`24` = "T cell", `25` = "pDC", `26` = "Proliferating Mast",`27` = "Neutrophils", `28` = "T1P")
levels(cd8.nhp.igg) <- c("Proliferating Macs","AM","SPP1 IM","TREM2 IM","IM","Monocytes","cDC1","cDC2","pDC","B Cells","Plasma Cells","Neutrophils","Proliferating T cells","T cell","NKT","NK","Proliferating Mast","Mast","Endothelial","Fibroblasts","T1P","T2P")
DotPlot(cd8.nhp.igg, features = c("MKI67","MAFB","ITGAX","PPARG","TREM2","SPP1","FLT3","CLEC9A","CLEC10A","CLEC4C","CD79B","MZB1","CXCR2","CD3G","KLRD1","FCER1A","CD34","COL1A1","EPCAM","SLC1A1","SFTPB"))

#Supp. Fig. 3b  - human.tb from Human TB Lung.R
human.tb <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Human TB resected lung/human.tb")
levels(human.tb) <- c("stromal","Mast","plasma cell","B cell","pDCs","mregDC","cDC2","cDC1","gd T","ILC","NK","CD8 T","Treg","CD4 T","Cycling T","SPP1 IM","FOLR2 IM","TREM2 IM","IM","monocytes","Cycling M")
DotPlot(human.tb, features = c("MKI67","MAFB","FCGR3A","CD14","ITGAX","TREM2","FOLR2","SPP1","CD3G","CD4","CD8A","FOXP3","KLRD1","IL7R","TRGV9","FLT3","CLEC9A","CLEC10A","CD200","CLEC4C","CD79B","MZB1","FCER1A","EPCAM"))

#Supp. Fig. 4 - human.tb from Human TB Lung.R; venn diagram made in Adobe Illustrator 
human.tb <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/Human TB resected lung/human.tb")
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
Spp1.Sp140 <- FindMarkers(super.infected.IM_filtered, ident.1 = "Sp140_ISG IM", ident.2 = "Sp140_Activated IM")
Spp1.Irg1 <- FindMarkers(super.infected.IM_filtered, ident.1 = "Irg1_ISG IM", ident.2 = "Irg1_Activated IM")
Spp1.iNOS <- FindMarkers(super.infected.IM_filtered, ident.1 = "iNOS_ISG IM", ident.2 = "iNOS_Activated IM")
Spp1.human <- FindMarkers(human.tb, ident.1 = "SPP1 IM", ident.2 = "TREM2 IM")

#Find the unique and conserved unique genes expressed by SPP1+ IMs
Spp1.Sp140.filt <- Spp1.Sp140 %>% filter(p_val_adj < 0.05 & avg_log2FC > 1) #72 genes
Spp1.Irg1.filt <- Spp1.Irg1 %>% filter(p_val_adj < 0.05 & avg_log2FC > 1) #107 genes
Spp1.iNOS.filt <- Spp1.iNOS %>% filter(p_val_adj < 0.05 & avg_log2FC > 1) #70 genes
Sp140.Irg1.overlap <- intersect(row.names(Spp1.Sp140.filt),row.names(Spp1.Irg1.filt)) #27 genes
Sp140.iNOS.overlap <- intersect(row.names(Spp1.Sp140.filt),row.names(Spp1.iNOS.filt)) #16 genes
Irg1.iNOS.overlap <- intersect(row.names(Spp1.Irg1.filt),row.names(Spp1.iNOS.filt)) #47 genes
triple.overlap <- intersect(Sp140.iNOS.overlap, Irg1.iNOS.overlap) #16 genes

Spp1.human.filt <- Spp1.human %>% filter(p_val_adj < 0.05 & avg_log2FC > 1) #1344 genes
Spp1.human.genes <-convert_human_to_mouse_symbols(row.names(Spp1.human.filt)) %>% na.omit() #1166 genes

Sp140.human.overlap <- intersect(row.names(Spp1.Sp140.filt),Spp1.human.genes) #19 genes
Irg1.human.overlap <- intersect(row.names(Spp1.Irg1.filt),Spp1.human.genes) #24 genes
iNOS.human.overlap <- intersect(row.names(Spp1.iNOS.filt),Spp1.human.genes) #14 genes
Sp140.Irg1.human.overlap <- intersect(Sp140.Irg1.overlap,Spp1.human.genes) #10 genes
iNOS.Irg1.human.overlap <- intersect(Irg1.iNOS.overlap,Spp1.human.genes) #10 genes
quad.overlap <- intersect(triple.overlap,Spp1.human.genes) #6 genes

#Genes conserved among TREM2+ IMs
Trem2.Sp140.filt <- Spp1.Sp140 %>% filter(p_val_adj < 0.05 & avg_log2FC < -1) #409
Trem2.Irg1.filt <- Spp1.Irg1 %>% filter(p_val_adj < 0.05 & avg_log2FC < -1) #491 genes
Trem2.iNOS.filt <- Spp1.iNOS %>% filter(p_val_adj < 0.05 & avg_log2FC < -1) #372 genes
Trem2.Sp140.Irg1.overlap <- intersect(row.names(Trem2.Sp140.filt),row.names(Trem2.Irg1.filt)) #121 genes
Trem2.Sp140.iNOS.overlap <- intersect(row.names(Trem2.Sp140.filt),row.names(Trem2.iNOS.filt)) #94 genes
Trem2.Irg1.iNOS.overlap <- intersect(row.names(Trem2.Irg1.filt),row.names(Trem2.iNOS.filt)) #218 genes
Trem2.triple.overlap <- intersect(Trem2.Sp140.iNOS.overlap, Trem2.Irg1.iNOS.overlap) #78 genes

#Supp. Fig. 5a
Idents(super.integrated) <- cell_type.state
levels(super.integrated) <- c("B6_naive","B6_Mtb-","B6_Mtb+","Sp140_naive","Sp140_Mtb-","Sp140_Mtb+","Irg1_naive","Irg1_Mtb-","Irg1_Mtb+","iNOS_naive","iNOS_Mtb-","iNOS_Mtb+")
VlnPlot(super.integrated, features = c("Il1r2","Il1rn"))

#Supp. Fig. 5b
FeaturePlot(super.integrated, features = "Il1rn", split.by = "state")

#Supp. Fig. 5c - dds.TBDM generated in Human TB Signatures_clean.R
dds <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/dds.TBDM")

#Graph plots
graph <- function(x){
  d <- plotCounts(dds, gene= x, intgroup="TB.status", returnData=TRUE)
  ggplot(d, aes(x = TB.status, y = count, color = TB.status)) + 
    geom_boxplot(coef=0, outlier.shape = NA) +
    geom_jitter(size = 3, alpha = 0.9) +
    theme_bw() +
    theme(
      legend.position = "none",
      plot.title = element_text(size=11)
    ) +
    ggtitle(x) +
    xlab("")
}

#IL1RN
graph("ENSG00000136689")

#Supp. Fig. 5d - tb.counts.norm and TB.infecfted are from Mouse to Human Signatures Clean.R
tb.counts.norm <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/tb.counts.norm")
TB.infected <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Human TB Gene Signatures/TB.infected")

immunosuppresion <- list(c("IL1RN","IL18BP","CD274","TGFB1","IDO1","IL10","ARG1"))
check_sig(tb.counts.norm, immunosuppresion)
immunosuppresion.scored <- hack_sig(tb.counts.norm, signatures = immunosuppresion, method = "ssgsea")
immunosuppresion.scored$status <- ifelse(grepl(paste(TB.infected, collapse = "|"), immunosuppresion.scored$sample_id), "TB","Uninfected")
immunosuppresion.roc <- roc(immunosuppresion.scored$status, immunosuppresion.scored$sig1, levels = c("TB","Uninfected")) #IFNg zscore AUC 83.8 ssgsea AUC 91.6 singscore AUC 88.7
plot(immunosuppresion.roc, print.auc = TRUE, col = "red", print.auc.y =0.47)

#Fig. 4a
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
#remove weirdo outliers
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "celltype"
DimPlot(super.infected.IM_filtered, label = T)
FeaturePlot(super.infected.IM_filtered, features = c("Cd9","Cd63","Itgax"))

#Fig. 4g
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
#remove weirdo outliers
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "celltype"
DimPlot(super.infected.IM_filtered, split.by = "state", label = T)

#Fig. 4i
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
#remove weirdo outliers
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "celltype"
FeaturePlot(super.infected.IM_filtered, features = c("Nos2","Arg1"), split.by = "cell_type")

#Supp. Fig. 6a
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
#remove weirdo outliers
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "celltype"
DimPlot(super.infected.IM_filtered, label = T)

#Supp. Fig. 6b
Idents(super.integrated) <- "infected"
super.infected <- subset(super.integrated, idents = "infected")
Idents(super.infected) <- "celltype"
super.infected.IM <- subset(super.infected, idents = c("Activated IM","ISG IM","IM","Mono","CD16-2 Mono"))
#remove weirdo outliers
cells.remove <- c("GATGATCTCCTAGCCT_4_3","GTCGTAACACTTGGCG_4_3","CACAGGCTCTGAGCAT_6_6","GTAGGTTGTGGATCAG_6_6","ATAGAGACAAGGCGTA_1_7")
super.infected.IM_filtered <- super.infected.IM[,!colnames(super.infected.IM) %in% cells.remove]
super.infected.IM_filtered <- RenameIdents(super.infected.IM_filtered, 'ISG IM' = "Spp1+ IM", 'Activated IM' = "Trem2+ IM", 'CD16-2 Mono' = "CD16.2 Mono")
Idents(super.infected.IM_filtered) <- "celltype"
FeaturePlot(super.infected.IM_filtered, features = c("Ly6c2","Fcgr4","Cd63","Cd9"))

#Fig. 6b - Tcell.singlet is from Tcell IL-1.R
Tcell.singlet <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/T cell ILC IL-1 scRNA-seq 12-30-20/Tcellsinglet")
DimPlot(Tcell.singlet, reduction = "wnn.umap")

#Fig. 6c - Tcell.singlet is from Tcell IL-1.R
Tcell.singlet <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/T cell ILC IL-1 scRNA-seq 12-30-20/Tcellsinglet")
FeaturePlot(Tcell.singlet, reduction = "wnn.umap", features = "Il1r1")

#Fig. 6d - Tcell.singlet is from Tcell IL-1.R
Tcell.singlet <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/T cell ILC IL-1 scRNA-seq 12-30-20/Tcellsinglet")
Idents(Tcell.singlet) <- "celltype.BM"
Th17.volcano <- FindMarkers(Tcell.singlet, ident.1 = "Th17_WT", ident.2 = "Th17_IL-1RKO", logfc.threshold = 0)
Th17.volcano <- Th17.volcano[c(1,2)]
EnhancedVolcano(Th17.volcano,
                lab = rownames(Th17.volcano),
                x = 'avg_log2FC',
                y = 'p_val',
                title = "Th17 WT vs IL-1R KO",
                subtitle = "",
                pCutoff = 0.05,
                FCcutoff = 1,
                selectLab = Th17.ligands,
                drawConnectors = TRUE,
                arrowheads = FALSE,
                labSize = 10)

#Fig. 6e - ILC.singlet from ILC IL-1.R
ILC.singlet <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/T cell ILC IL-1 scRNA-seq 12-30-20/ILCsinglet")
DimPlot(ILC.singlet, reduction = "wnn.umap")

#Fig. 6f - ILC.singlet from ILC IL-1.R
ILC.singlet <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/T cell ILC IL-1 scRNA-seq 12-30-20/ILCsinglet")
FeaturePlot(ILC.singlet, reduction = "wnn.umap", features = "Il1r1")

#Fig. 6g - ILC.singlet from ILC IL-1.R
ILC.singlet <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/T cell ILC IL-1 scRNA-seq 12-30-20/ILCsinglet")
Idents(ILC.singlet) <- "celltype.BM"
ILC3.volcano <- FindMarkers(ILC.singlet, ident.1 = "ILC3_WT", ident.2 = "ILC3_IL-1RKO", logfc.threshold = 0)
ILC3.volcano <- ILC3.volcano[c(1,2)]
p1 <- EnhancedVolcano(ILC3.volcano,
                      lab = rownames(ILC3.volcano),
                      x = 'avg_log2FC',
                      y = 'p_val',
                      title = "ILC3 WT vs IL-1R KO",
                      subtitle = "",
                      pCutoff = 0.05,
                      FCcutoff = 0.75,
                      xlim = c(-5,5),
                      ylim = c(0,10))

NKT17.volcano <- FindMarkers(ILC.singlet, ident.1 = "NKT17_WT", ident.2 = "NKT17_IL-1RKO", logfc.threshold = 0)
NKT17.volcano <- NKT17.volcano[c(1,2)]
p2 <- EnhancedVolcano(NKT17.volcano,
                      lab = rownames(NKT17.volcano),
                      x = 'avg_log2FC',
                      y = 'p_val',
                      title = "NKT17 WT vs IL-1R KO",
                      subtitle = "",
                      pCutoff = 0.05,
                      FCcutoff = 0.75,
                      xlim = c(-5,5),
                      ylim = c(0,10))

#Fig. 6i - stromal.combined from Stromal Cell Mtb.R
stromal.combined <- readRDS(file = "/Users/DmitriKotov/library/CloudStorage/Box-Box/Coding stuff/Stromal Cell IL-1 scRNA-seq 7-22-21/stromalcombined")
DimPlot(stromal.combined)

#Fig. 6j - stromal.combined from Stromal Cell Mtb.R
stromal.combined <- readRDS(file = "/Users/DmitriKotov/library/CloudStorage/Box-Box/Coding stuff/Stromal Cell IL-1 scRNA-seq 7-22-21/stromalcombined")
FeaturePlot(stromal.combined, features = "Il1r1")

#Fig. 6k - stromal.combined from Stromal Cell Mtb.R
stromal.combined <- readRDS(file = "/Users/DmitriKotov/library/CloudStorage/Box-Box/Coding stuff/Stromal Cell IL-1 scRNA-seq 7-22-21/stromalcombined")
simplified.stromal <- RenameIdents(stromal.combined,'Myofibroblast 1' = "Myofibroblast",'Myofibroblast 2' = "Myofibroblast", 'Myofibroblast 3' = "Myofibroblast", 'Myofibroblast 4' = "Myofibroblast",
                                   'Fibromyo/SMC' = "Myofibroblast", 'Pericyte' = "Myofibroblast", 'Col13a1+ fibroblast 1' = "Fibroblast", 'Col13a1+ fibroblast 2' = "Fibroblast",
                                   'Col13a1+ fibroblast 3' = "Fibroblast",'Col13a1+ fibroblast 4' = "Fibroblast", 'Col14a1 fibroblast 1' = "Fibroblast", 'Col14a1 fibroblast 2' = "Fibroblast",
                                   'aCap 1' = "Endothelial", 'aCap 2' = "Endothelial", 'gCap 3' = "Endothelial", 'Esm1 Endothelial' = "Endothelial", 'Art' = "Endothelial", 'gCap 2' = "Endothelial",
                                   'gCap 1' = "Endothelial", 'Vein' = "Endothelial", 'Club 1' = "Club", 'Club 2' = "Club")
simplified.stromal$celltype_state_genotype <- paste(Idents(simplified.stromal),simplified.stromal$state, simplified.stromal$genotype, sep = "_")

#Look at differentially expressed genes
potential.ligands <- unique(lr_network$from) %>% convert_human_to_mouse_symbols() %>% na.omit

Idents(simplified.stromal) <- "celltype_state_genotype"
Myo.il1.markers <- FindMarkers(simplified.stromal, assay = "RNA", ident.1 = "Myofibroblast_Mtb_WT", ident.2 = "Myofibroblast_Mtb_IL1KO")
Myo.il1.volcano <- Myo.il1.markers[c(2,5)]
regulated.Myo <- Myo.il1.volcano %>% filter(avg_log2FC > 1 | avg_log2FC < -1)  %>% row.names()
Myo.ligands <- intersect(potential.ligands, regulated.Myo)

p1 <- EnhancedVolcano(Myo.il1.volcano,
                      lab = rownames(Myo.il1.volcano),
                      x = 'avg_log2FC',
                      y = 'p_val_adj',
                      title = "Mtb infected WT vs IL-1R KO Myofibroblast cells",
                      subtitle = "",
                      pCutoff = 0.05,
                      xlim = c(-5,5),
                      ylim = c(0,60),
                      selectLab = Myo.ligands,
                      drawConnectors = TRUE,
                      arrowheads = FALSE,
                      labSize = 10)

Fibro.il1.markers <- FindMarkers(simplified.stromal, assay = "RNA", ident.1 = "Fibroblast_Mtb_WT", ident.2 = "Fibroblast_Mtb_IL1KO")
Fibro.il1.volcano <- Fibro.il1.markers[c(2,5)]
regulated.Fibro <- Fibro.il1.volcano %>% filter(avg_log2FC > 1 | avg_log2FC < -1)  %>% row.names()
Fibro.ligands <- intersect(potential.ligands, regulated.Fibro)

p2 <- EnhancedVolcano(Fibro.il1.volcano,
                      lab = rownames(Fibro.il1.volcano),
                      x = 'avg_log2FC',
                      y = 'p_val_adj',
                      title = "Mtb infected WT vs IL-1R KO Fibroblast cells",
                      subtitle = "",
                      pCutoff = 0.05,
                      xlim = c(-5,5),
                      ylim = c(0,60),
                      selectLab = Fibro.ligands,
                      drawConnectors = TRUE,
                      arrowheads = FALSE,
                      labSize = 10)

Endo.il1.markers <- FindMarkers(simplified.stromal, assay = "RNA", ident.1 = "Endothelial_Mtb_WT", ident.2 = "Endothelial_Mtb_IL1KO")
Endo.il1.volcano <- Endo.il1.markers[c(2,5)]
regulated.Endo <- Endo.il1.volcano %>% filter(avg_log2FC > 1 | avg_log2FC < -1)  %>% row.names()
Endo.ligands <- intersect(potential.ligands, regulated.Endo)

p3 <- EnhancedVolcano(Endo.il1.volcano,
                      lab = rownames(Endo.il1.volcano),
                      x = 'avg_log2FC',
                      y = 'p_val_adj',
                      title = "Mtb infected WT vs IL-1R KO Endothelial cells",
                      subtitle = "",
                      pCutoff = 0.05,
                      xlim = c(-5,5),
                      ylim = c(0,60),
                      selectLab = Endo.ligands,
                      drawConnectors = TRUE,
                      arrowheads = FALSE,
                      labSize = 10)

p1 + p2 + p3

#Supp. Fig. 7a
DimPlot(super.integrated)

#Supp. Fig. 7b
FeaturePlot(super.integrated, features = "Il1r1", split.by = "state")

#Supp. Fig. 8a - Tcell.singlet is from Tcell IL-1.R
Tcell.singlet <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/T cell ILC IL-1 scRNA-seq 12-30-20/Tcellsinglet")
DotPlot(Tcell.singlet, features = c("Cd3e","Mki67","Cd4","Cd8a","Cx3cr1","Cxcr3","Sell","adt_CD44","adt_CXCR5","Il17a","Foxp3","Isg15"))

#Supp. Fig. 8b - ILC.singlet from ILC IL-1.R
ILC.singlet <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Coding stuff/T cell ILC IL-1 scRNA-seq 12-30-20/ILCsinglet")
DotPlot(ILC.singlet, features = c("Fcgr1","Mertk","Lyve1","Zbtb46","Cd79a","Cd3e","Mki67","Cd4","Cd8a","Sell","Cxcr3","adt_NKp46","Zbtb16","Il17a","Il7r","adt_ST2",
                                  "Klrg1","Il18r1","Rorc","Il22"))

#Supp. Fig. 8c - stromal.combined from Stromal Cell Mtb.R
stromal.combined <- readRDS(file = "/Users/DmitriKotov/library/CloudStorage/Box-Box/Coding stuff/Stromal Cell IL-1 scRNA-seq 7-22-21/stromalcombined")
DotPlot(stromal.combined, features = c("Lyve1","Pecam1","Epcam","Col1a1","Ptprc","Msln","Sftpc","Akap5","Scgb1a1","Col14a1","Col13a1","Tgfbi","Notch3","Car4","Gpihbp1","Bmx","Slc6a2","Esm1","Ly6g","Ccr2","Fcgr1","Itgax","Ear1","Cd3e","Sell","Cd4","Cd8a","Cd79a","Gzma"))

#Supp. Fig. 9b - p_ligand_receptor_network is from Nichnet on IL-1 Regulated Genes.R
p_ligand_receptor_network <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Irg1 and iNOS scRNAseq 220315/p_ligand_receptor_network")
p_ligand_receptor_network

#Supp. Fig. 9c - p_ligand_aupr_Activated and p_ligand_aupr_ISG are from Nichnet on IL-1 Regulated Genes.R
p_ligand_aupr_Activated <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Irg1 and iNOS scRNAseq 220315/p_ligand_aupr_Activated")
p_ligand_aupr_ISG <- readRDS(file = "/Users/dmitrikotov/Library/CloudStorage/Box-Box/Dmitri Data/Coding stuff/Irg1 and iNOS scRNAseq 220315/p_ligand_aupr_ISG")
p_ligand_aupr_Activated
p_ligand_aupr_ISG